servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingContentCodec.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Publisher;
+
+/**
+ * API for HTTP <a href="https://tools.ietf.org/html/rfc7231#section-3.1.2.1">Content Codings</a>.
```

Comment:
Since the codec classes do not carry any http-related information it's better to avoid mentioning HTTP here. Adding this reference in `encoding()` methods on `HttpMetaData` is enough.
We have a potential to move them to another module outside of HTTP. Not sure if we need to do it now.